docs/SPATIAL_FILTERING.md

```diff
@@ -0,0 +1,109 @@
+-----------------
+
+Starting with Kart 0.11.0, Kart supports spatially-filtered repositories. This is a Kart repository with an extra attached geometry describing a region that the user is interested in - any features that intersect that geometry are said to match the spatial filter, and they are shown. Conversely any features that do not intersect with it do not
+match, and are hidden.
+
+Spatial filtering has two main benefits:
+
+- Saves time, bandwidth, and disk space when cloning a repo and creating a working copy, by only downloading matching features.
+- Saves time loading the working copy into an editor and saves time for the user to focus in on their area of interest.
+
+The spatial filter is applied in three different ways:
+
+1. Only features that match the spatial filter are cloned during a `kart clone`.
+2. Kart commands that output features only output the features that match the spatial filter.
+3. The working copy is only populated with features that match the spatial filter.
+
+All three of these ways have some additional complexity, which will be explained in the following three sections. If this seems like more detail than you will need, skip ahead to learn how to [set the spatial filter](#Setting-the-spatial-filter)
+
+### During clone
+
+Spatial filtering during a `kart clone` is only performed if the spatial filter is supplied up front as part of the clone command. The spatial filter can be changed any time after the clone operation is complete, but at this point the entire repository will already have been cloned and it will be too late to save on download time and bandwidth. Note that none of the cloned data is deleted when changing the spatial filter anytime after cloning. To put this another way:
+
+1. During a clone, a spatial filter can be supplied. Only features that match this original spatial filter are cloned. The set of features cloned remains the same from this point onwards.
+2. Kart commands only output the features that match the *current* spatial filter, which can be changed at any time.
+3. The working copy is only populated with features that match the *current* spatial filter, which can be changed at any time.
+
+As a result, a current limitation is that the spatial filter can be changed at any time, but only to a subset of the filter that was used during the clone. If you need to grow the spatial filter beyond what was originally cloned, the workaround currently is to start again with a fresh clone.
+
+There are two more qualifications regarding spatial filtering during the clone operation -
+- It is approximate or "best-effort" - some features may be downloaded that are not actually needed. However, since the spatial-filter is applied precisely in the other two stages (Kart command output and working copy creation) this will be hidden from you. The cloned data on disk may contain any number of features that are outside the spatial filter without any adverse effects apart from reduced efficiency.
```

Comment:
```suggestion
There are two more qualifications regarding spatial filtering during clone and fetch operations -
- All features that are needed will be downloaded, but some features may be downloaded that are not actually needed. Since the spatial-filter is applied precisely in the other two stages (Kart command output and working copy creation) any extra features will be hidden from you. The cloned data on disk may contain any number of features that are outside the spatial filter without any adverse effects.
```